modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLCopyFolderDisplayContext.java

```diff
@@ -162,12 +166,22 @@ private String _getItemSelectedEventName() {
 		return portletDisplay.getNamespace() + "folderSelected";
 	}
 
+	private long _getSourceFolderId() {
+		if (_sourceFolderId < 0) {
+			_sourceFolderId = ParamUtil.getLong(
+				_httpServletRequest, "sourceFolderId");
+		}
+
+		return _sourceFolderId;
+	}
+
+	private long[] _entryIds;
+	private String _entryName;
 	private final HttpServletRequest _httpServletRequest;
 	private final LiferayPortletResponse _liferayPortletResponse;
 	private String _redirect;
-	private long _sourceFolderId = -1;
-	private String _sourceFolderName;
-	private long _sourceRepositoryId = -1;
```

Comment:
Removing the default value of `-1` means:

* For `_sourceRepositoryId` probably nothing, because I think there is no default repo with value `0`
* For `_sourceFolderId` means that it's never updated because we are still checking for negative value in `_getSourceFolderId` (we cannot check for zero value because it's a valid value). So, at the end we have that the folder selector for the destination folder always open at root folder, and not at the current subfolder.